City search function

mrvanagas opened the pull request:
Added city search page and component, by entering part of city name, gets a drop down menu and shows a 7 day forecast 
![image](https://user-images.githubusercontent.com/71028004/168264530-26b33bc2-45c6-4426-b804-57891fa4ba00.png)

To do - improve styling, added icon handling from API.

mrvanagas committed:
basic api functions

Added 3 basic API functions, also gets local coordinates

mrvanagas committed:
basic API functions

Basic API functions with 3 calls

mrvanagas committed:
Project restructure, added server

Restructured project structure to separate folders, added backend server to generate tokens.

mrvanagas committed:
removed token

Removed hardcoded token

mrvanagas committed:
Adjusted geolocator function

Geolocator now is called automatically in API.js file to get current current coordinates

Commit:
Auto weather info

Added logic to automatically show weather, based on current position, using the geolocator

Commit:
code modification

mrvanagas committed:
Removed hardcoded token, added other pages and router

Removed hard coded token, added several pages as placeholders, added router

mrvanagas committed:
Removed token

Removed hardcoded token

mrvanagas committed:
added 1st iteration of silent auth

Added first iteration of silent authentication, exploring other options with timed tokens

mrvanagas committed:
fixed files

fixed wrong file deletion

mrvanagas committed:
first iteration of search

First iteration of search functionality, first working with static data, in order to make the foundation of the search, will change to getting the info from API

mrvanagas committed:
adjustments

mrvanagas committed:
Added styled components

Refactored code to use styled components instead of regular css

mrvanagas committed:
added search from API

Added search from API, now need to figure out how to pass city id to the end point

mrvanagas committed:
removed unneeded file

mrvanagas committed:
Added city search

Added city search by entering part of the city name.

mrvanagas commented:
Fixed destination branch

JuliaShavyrova commented:
please resolve conflicts first

mrvanagas committed:
Merge branch 'Vanagas_Lukas' into city_search_function

JuliaShavyrova commented on client/src/Components/MultiCard.styles.js:
```diff
@@ -0,0 +1,31 @@
+    justify-content: space-between;
+    align-items: center;
+    align-content: center;
+    margin: 1rem 1rem 1rem 1rem;
```
style can be written shorter

mrvanagas replied:
Shortened style

JuliaShavyrova commented on client/src/Components/MultiCard.styles.js:
```diff
@@ -0,0 +1,31 @@
+
+export const CardContainer = styled.div`
+  background-color: gray;
+  border-radius: 0.5rem 0.5rem 0.5rem 0.5rem;
```
the same

mrvanagas replied:
Shortened style

mrvanagas commented:
> please resolve conflicts first

Done, conflicts resolved.

mrvanagas committed:
Added dateHelper

Added datehelper, to convert date to weekdays, shortened styles.

mrvanagas committed:
Merge branch 'city_search_function' of https://github.com/mrvanagas/ReactTask into city_search_function

mrvanagas committed:
feat: added date formating

Added dateHelper file, to format the full date into weekday.

mrvanagas committed:
feat: added first iteration of debouncing

Added first iteration of debouncing the search in the city search page.

JuliaShavyrova commented on client/src/Components/SearchBar.jsx:
```diff
@@ -0,0 +1,105 @@
+
+    if (inputValue.length > minCharacters) {
+      console.log('start of request')
+     timerID = setTimeout(async () =>{
```
seems like your code needs auto-formatting (incorrect spacing, brackets, etc)

mrvanagas replied:
Code should be formatted now

mrvanagas committed:
code clean up

mrvanagas commented:
Added debouncing and cleaned up code

JuliaShavyrova commented on client/src/API/api.js:
```diff
@@ -11,7 +11,7 @@ export async function getTokenFromAPI() {
   try {
     const response = await axios.get(TOKEN_URL);
     console.log(response);
```
please remove console logs if they are not important for user

mrvanagas replied:
Removed logs in a few files, will do another sweep.

JuliaShavyrova commented on client/src/Components/MultiCard.jsx:
```diff
@@ -0,0 +1,22 @@
+const MultiWeather = ({ data }) => {
+  return (
+    <Style.Container>
+      {data.map((info, index) => (
```
seems like `info` can be destructured

mrvanagas replied:
Done

JuliaShavyrova commented on client/src/Utils/dateHelper.js:
```diff
@@ -0,0 +1,17 @@
+function dateHelper(date) {
```
please rename function, it's not clear what exactly function helps with

mrvanagas replied:
Renamed

JuliaShavyrova commented on client/src/Utils/dateHelper.js:
```diff
@@ -0,0 +1,17 @@
+    6: 'Sunday'
+  };
+
+  return weekDayMap[weekDay];
```
function can be simplified and implemented with no hardcoded array

mrvanagas replied:
Done, refactored.

mrvanagas committed:
code changes

Formatted code in SearchBar, destructured `info` in MultiCard, removed unneeded console.logs in files. Still need to refactor dateHelper file.

mrvanagas committed:
dateHelper refactor

Refactored dateHelper function, shortened it.

JuliaShavyrova commented on client/src/Components/SearchBar.jsx:
```diff
@@ -0,0 +1,80 @@
+          alert(error);
+        }
+      }
+        , 900)
```
still see incorrect formatting, please use npm scripts and packages provided in your project folder

mrvanagas replied:
Ran code formatter, should be now properly formatted

JuliaShavyrova commented on client/src/Utils/dateHelper.js:
```diff
@@ -0,0 +1,7 @@
+function dateToWeekdayHelper(date) {
+  const weekDay = new Date(date).toLocaleString('en-US', { weekday: 'long' });
+
+  return weekDay
```
no need to create variable just for returning it in next line

mrvanagas replied:
Changed

mrvanagas committed:
code refactor

Shortened dateHelper, ran code formatter, should now be properly formatted
